File: app/src/main/java/net/mbonnin/arcanetracker/CardUtil.java
package net.mbonnin.arcanetracker;


import net.mbonnin.hsmodel.Card;
import net.mbonnin.hsmodel.CardJson;

import java.util.Collections;

public class CardUtil {
    public static final Card UNKNOWN = unknown();

    public static Card unknown() {
        Card card = new Card();
        card.name = "?";
        card.playerClass = "?";
        card.cost = Card.UNKNOWN_COST;
        card.id = "?";
        card.rarity = "?";
        card.type = Card.TYPE_UNKNOWN;
        card.text = "?";
        card.race = "?";
        card.collectible = false;
        return card;
    }

    public static Card secret(String clazz) {
        Card card = unknown();
        card.type = Card.TYPE_SPELL;
        card.text = Utils.getString(R.string.secretText);
        int classIndex = Card.Companion.niceNameToClassIndexNC(clazz);
        switch (classIndex) {
            case Card.CLASS_INDEX_HUNTER:
                card.id = "secret_h";
                card.cost = 2;
                card.playerClass = Card.CLASS_HUNTER;
                break;
            case Card.CLASS_INDEX_MAGE:
                card.id = "secret_m";
                card.cost = 3;
                card.playerClass = Card.CLASS_MAGE;
                break;
            case Card.CLASS_INDEX_PALADIN:
                card.id = "secret_p";
                card.cost = 1;
                card.playerClass = Card.CLASS_PALADIN;
                break;
        }
        card.name = Utils.getString(R.string.secret);
        return card;
    }

    public static Card getCard(int dbfId) {

        for (Card card: CardJson.INSTANCE.allCards()) {
            if (card.dbfId == dbfId) {
                return card;
            }
        }

        return null;
    }

    public static Card getCard(String key) {
        int index = Collections.binarySearch(CardJson.INSTANCE.allCards(), key);
        if (index < 0) {
            return UNKNOWN;
        } else {
            return CardJson.INSTANCE.allCards().get(index);
        }
    }
}

File: app/src/main/kotlin/net/mbonnin/arcanetracker/CardUtil.kt
package net.mbonnin.arcanetracker


import net.hearthsim.hsmodel.Card
import net.hearthsim.hsmodel.CardJson
import net.hearthsim.hsmodel.enum.HSSet
import net.hearthsim.hsmodel.enum.PlayerClass
import net.hearthsim.hsmodel.enum.Rarity
import net.hearthsim.hsmodel.enum.Type

object CardUtil {
    fun secret(playerClass: String): Card {
        val id: String
        val cost: Int
        val pClass: String

        when (playerClass) {
            PlayerClass.PALADIN -> {
                id = "secret_p"
                cost = 1
                pClass = PlayerClass.PALADIN
            }
            PlayerClass.HUNTER -> {
                id = "secret_h"
                cost = 2
                pClass = PlayerClass.HUNTER
            }
            PlayerClass.ROGUE -> {
                id = "secret_r"
                cost = 2
                pClass = PlayerClass.ROGUE
            }
            PlayerClass.MAGE -> {
                id = "secret_m"
                cost = 3
                pClass = PlayerClass.MAGE
            }
            else -> return CardJson.unknown()
        }

        return Card(
                type = Type.SPELL,
                text = Utils.getString(R.string.secretText),
                name = Utils.getString(R.string.secret),
                id = id,
                cost = cost,
                playerClass = pClass,
                dbfId = 0,
                set = HSSet.CORE
        )
    }


    fun getCard(key: String): Card {
        return ArcaneTrackerApplication.get().cardJson.getCard(key)
    }


    fun getDust(rarity: String?, golden: Boolean): Int {
        if (rarity == null) {
            return 0
        }
        return when (rarity) {
            Rarity.COMMON -> if (golden) 50 else 5
            Rarity.RARE -> if (golden) 100 else 20
            Rarity.EPIC -> if (golden) 400 else 100
            Rarity.LEGENDARY -> if (golden) 1600 else 400
            else -> 0
        }
    }
}
